### PR TITLE
Add Fortran OpenACC/CUDA example

### LINKED ISSUE #1403: Request for Example: Fortran+CUDA
We are preparing a hands-on example for our [ESM User Forum](https://www.fz-juelich.de/SharedDocs/Termine/IAS/JSC/EN/courses/2020/esm-user-forum-2020.html) in a few days and want to show usage of GridTools for that occasion.

We would like to use the Fortran binding capabilities to launch a GridTools 2D Jacobi stencil on the GPU.

As a GridTools newbie, I lack an example, showing Fortran usage of GPU GridTools stencils. I found the [`driver.f90` of the `c_bindings`](https://github.com/GridTools/gridtools/blob/master/examples/c_bindings/driver.f90) example, but that uses the MC backend. Is there one which I overlooked? Or can you quickly put something together which I can use as a blueprint?

We intend to use either GCC as our host compiler or PGI (we are showing an OpenACC stencil as well).

### PR DESCRIPTION
**Pull request by havogt**
Additional changes:
- `is_gpu_ptr()` returns true also if ptr is a managed memory ptr.

Fixes #1403.

### CODE REVIEW

**Comment by havogt** [comment]
TODO also port to master

**Comment by havogt** [comment]
launch strgrid

File: examples/c_bindings/driver_cuda.f90 [hunk 1]
```diff
@@ -0,0 +1,58 @@
+! GridTools
+!
+! Copyright (c) 2014-2019, ETH Zurich
```
Comment (mbianco) [inline examples/c_bindings/driver_cuda.f90]: 2020?

File: examples/c_bindings/driver_cuda.f90 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    in_handle = make_data_store(storage_info_handle)
+    out_handle = make_data_store(storage_info_handle)
+    computation_handle = make_copy_stencil(grid_handle)
+    ! bindgen_handles need to be released explicitly
+    call bindgen_release(grid_handle)
+    call bindgen_release(storage_info_handle)
+
+    ! transform data from Fortran to C layout
+    !$acc data copy(out_array,in_array)
+    call transform_f_to_c(in_handle, in_array)
```
Comment (mbianco) [inline examples/c_bindings/driver_cuda.f90]: I guess there are missing files in the PR

Comment (havogt) [reply examples/c_bindings/driver_cuda.f90]: No I don't think so? Doesn't it work for you?

File: examples/c_bindings/driver_cuda.f90 (hunk 1 above)
Comment (havogt) [reply examples/c_bindings/driver_cuda.f90]: What would be our policy here? Update on a file-touched basis or do a full sweep now?

File: examples/c_bindings/driver_cuda.f90 (hunk 2 above)
Comment (havogt) [reply examples/c_bindings/driver_cuda.f90]: It's the same transform that we use in the c code or the fortran host version.

File: examples/c_bindings/driver_cuda.f90 (hunk 1 above)
Comment (mbianco) [reply examples/c_bindings/driver_cuda.f90]: not sure. I principle only the first date matters

File: examples/c_bindings/driver_cuda.f90 (hunk 2 above)
Comment (mbianco) [reply examples/c_bindings/driver_cuda.f90]: I messed up something, sorry... it works

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by AndiH** [comment]
FYI: I wanted to add some debug info into the whole process, and also fine-tune the OpenACC compiler flags, so I moved from

```cmake
 set(OpenACC_Fortran_FLAGS "-acc")
```

to

```cmake
 set(OpenACC_Fortran_FLAGS "-acc -ta=tesla:cc30 -Minfo=accel")
```

– but it fails!

It works when NOT quoting the flags. Apparently, the quotes are passed through to the compiler and it thinks that this is one large flag – which it cannot parse:

```cmake
 set(OpenACC_Fortran_FLAGS -acc -ta=tesla:cc30 -Minfo=accel)
```

File: examples/c_bindings/CMakeLists.txt.in [hunk 3]
```diff
@@ -69,5 +69,29 @@ if (CMAKE_CUDA_COMPILER)
             target_link_libraries(example_driver_cuda_c copy_stencil_lib_cuda_c)
             add_test(NAME example_driver_cuda_c COMMAND $<TARGET_FILE:example_driver_cuda_c>)
         endif()
+
+        if (CMAKE_Fortran_COMPILER_LOADED)
+            if(CMAKE_Fortran_COMPILER_ID MATCHES "PGI")
+                # FindOpenACC seems to be broken for modern versions of PGI, we assume PGI has OpenACC...
+                set(OpenACC_Fortran_FLAGS "-acc")
```
Comment (havogt) [inline examples/c_bindings/CMakeLists.txt.in]:
```suggestion
                set(OpenACC_Fortran_FLAGS -acc)
```

**Comment by havogt** [comment]
Thanks @AndiH . That's probably one of the biggest CMake issue. You never know which command expects a string and which a list...

**Comment by havogt** [comment]
Will go to master in #1454, won't backport to v1.1.

**Pull request closed without merging**